crates/uv-torch/src/lib.rs

```diff
@@ -0,0 +1,413 @@
+}
+
+impl TorchStrategy {
+    /// Determine the [`TorchStrategy`] from the given [`TorchMode`], [`Os`], and [`Accelerator`].
+    pub fn from_mode(mode: TorchMode, os: &Os, accelerator: Option<&Accelerator>) -> Self {
+        match mode {
+            TorchMode::Auto => {
+                if let Some(Accelerator::Cuda { driver_version }) = accelerator {
+                    Self::Auto {
+                        os: os.clone(),
+                        driver_version: driver_version.clone(),
+                    }
+                } else {
+                    Self::Backend(TorchBackend::Cpu)
+                }
+            }
+            TorchMode::Cpu => Self::Backend(TorchBackend::Cpu),
+            TorchMode::Cu126 => Self::Backend(TorchBackend::Cu126),
+            TorchMode::Cu125 => Self::Backend(TorchBackend::Cu125),
+            TorchMode::Cu124 => Self::Backend(TorchBackend::Cu124),
+            TorchMode::Cu123 => Self::Backend(TorchBackend::Cu123),
+            TorchMode::Cu122 => Self::Backend(TorchBackend::Cu122),
+            TorchMode::Cu121 => Self::Backend(TorchBackend::Cu121),
+            TorchMode::Cu120 => Self::Backend(TorchBackend::Cu120),
+            TorchMode::Cu118 => Self::Backend(TorchBackend::Cu118),
+            TorchMode::Cu117 => Self::Backend(TorchBackend::Cu117),
+            TorchMode::Cu116 => Self::Backend(TorchBackend::Cu116),
+            TorchMode::Cu115 => Self::Backend(TorchBackend::Cu115),
+            TorchMode::Cu114 => Self::Backend(TorchBackend::Cu114),
+            TorchMode::Cu113 => Self::Backend(TorchBackend::Cu113),
+            TorchMode::Cu112 => Self::Backend(TorchBackend::Cu112),
+            TorchMode::Cu111 => Self::Backend(TorchBackend::Cu111),
+            TorchMode::Cu110 => Self::Backend(TorchBackend::Cu110),
+            TorchMode::Cu102 => Self::Backend(TorchBackend::Cu102),
+            TorchMode::Cu101 => Self::Backend(TorchBackend::Cu101),
+            TorchMode::Cu100 => Self::Backend(TorchBackend::Cu100),
+            TorchMode::Cu92 => Self::Backend(TorchBackend::Cu92),
+            TorchMode::Cu91 => Self::Backend(TorchBackend::Cu91),
+            TorchMode::Cu90 => Self::Backend(TorchBackend::Cu90),
+            TorchMode::Cu80 => Self::Backend(TorchBackend::Cu80),
+        }
+    }
+
+    /// Return the appropriate index URLs for the given [`TorchStrategy`] and [`PackageName`].
+    pub fn index_urls(
+        &self,
+        package_name: &PackageName,
+    ) -> Option<impl Iterator<Item = &IndexUrl>> {
+        if !matches!(
+            package_name.as_str(),
+            "torch"
+                | "torch-model-archiver"
+                | "torch-tb-profiler"
+                | "torcharrow"
+                | "torchaudio"
+                | "torchcsprng"
+                | "torchdata"
+                | "torchdistx"
+                | "torchserve"
+                | "torchtext"
+                | "torchvision"
+                | "pytorch-triton"
```

Comment:
Can we add this list to some documentation? Reading the high-level overview I didn't realize we were hardcoding a package list.

Reply:
Can we generate this by querying the PyTorch indices to see what they have? (Maybe a manually-run script that queries them and updates this list, or an automatically-run integration tests that makes sure this list is in sync with what's currently on their indices?)

Along those lines it would be helpful to have this list somewhere declarative. It might also be helpful to allow user-controlled overrides of this list if the set of packages changes.

Reply:
Unfortunately I don't know that we can... We don't want _all_ packages on these indexes, because they include things like `jinja2`. And in some cases, they include _incomplete_ packages like `markupsafe` (where they only have a few wheels).